ml-agents/mlagents/trainers/learn.py

```diff
@@ -133,9 +131,8 @@ def run_training(run_seed: int, options: RunOptions) -> None:
         maybe_meta_curriculum = try_create_meta_curriculum(
             options.curriculum, env_manager, restore=checkpoint_settings.resume
         )
-        sampler_manager, resampling_interval = create_sampler_manager(
-            options.parameter_randomization, run_seed
-        )
+        maybe_add_samplers(options.parameter_randomization, env_manager)
```

Comment:
Currently, samplers are only added once. I can extend this PR to allow trainers to send new sampling encodings over the course of an experiment.